src/html_support_files/odoc.css

```diff
@@ -754,11 +773,217 @@ td.def-doc *:first-child {
+  display: none;
+}
+
+
+.odoc-search .search-result {
+  background: var(--toc-background);
+  position: absolute;
+  left: 0;
+  right: 0;
+}
+
+.odoc-search .search-result-inner {
+  border: solid;
+  border-color: var(--search-results-border);
+  border-width: 1px;
+  margin-top: 8px;
+  border-radius: 6px;
+  box-shadow: 0 3px 10px 2px var(--search-results-shadow);
+  max-height: 40rem;
+  overflow-y: auto;
+}
+
+.search-bar {
+  /* inputs are of fixed size by default, even if you display:block them */
+  width: 100%;
+}
+
+.odoc-search .search-entry {
+  color: var(--color);
+  display: grid;
+  /* This constant of 59px was carefully chosen to accomodate every possible
+     "kind". We cannot ask CSS to compute it because not every kind will be
+     present in every search result list, and we do not want the size of the
+     column to change with different results. */
+  grid-template-columns: [kinds] 59px [titles] 1fr;
+  flex-wrap: nowrap;
+  flex-direction: row;
+  border-bottom: var(--search-results-border) solid 1px;
+  background-color: inherit;
+  outline:0;
+  padding-right: 0.5rem;
+
+}
+
+.odoc-search .search-entry p {
+  margin: 0;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+.odoc-search .search-entry:focus-visible {
+  box-shadow: none;
+  background-color: var(--target-background);
+}
+
+.odoc-search .search-entry:hover {
+  box-shadow: none;
+  background-color: var(--toc-background-emph);
+}
+
+.odoc-search .search-entry .entry-kind {
+  grid-row: 1/2;
+  grid-column: 1/2;
+  line-height: 1.4rem;
+  padding-bottom: 0.25rem;
+  padding-top: 0.5rem;
+  font-size: 0.8em;
+  font-weight: bold;
+  margin-right: 0.5rem;
+  text-align: right;
+  position:relative;
+  bottom: 0;
+}
+
+.odoc-search .search-entry pre {
+  border: none;
+  margin: 0;
+}
+
+.odoc-search .search-entry pre code {
+  font-size: 1em;
+  background-color: var(--li-code-background);
+  color: var(--li-code-color);
+  border-radius: 3px;
+  padding: 0 0.3ex;
+}
+
+.odoc-search .search-entry .entry-title {
+  width: 100%;
+  display: block;
+  grid-column: 2/2;
+  grid-row: 1/2;
+  align-self: end;
+  line-height: 1.4rem;
+  padding-bottom: 0.25rem;
+  padding-top: 0.5rem;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.odoc-search .entry-name {
+  font-weight: bold;
+}
+
+.odoc-search .prefix-name {
+  font-weight: bold;
+}
+
+.odoc-search .search-entry .prefix-name {
+  opacity: 0.7;
+}
+
+.odoc-search .entry-rhs {
+  white-space: nowrap;
+}
+
+.odoc-search .search-entry .entry-content {
+  flex-grow: 1;
+  flex-shrink: 1;
+  min-width: 0;
+}
+
+.odoc-search .search-entry .entry-comment {
+  max-height: 1.5em;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  font-size: 0.95em;
+  grid-row: 2/2;
+  grid-column: 2/2;
+  margin-bottom: 0.5rem;
+}
+
+.odoc-search .search-entry .entry-comment ul {
+  white-space: nowrap;
+  display: inline;
```

Comment:
What is the goal of these ? (which is repeated a few times and also set on parent element)

Reply:
The goal of this is to ensure that everything is displayed on one line until the ellipsis.

Reply:
Is it needed even if it's specified on the parent ?

Reply:
Yeses, because the style of the paragraph or the code block would override the style of the parent. Its a bit weird, I did not expect this at first either.